includes/woopay/class-woopay-session.php

```diff
@@ -147,11 +147,11 @@ public static function get_user_id_from_cart_token() {
 
 			// If the email is verified on WooPay and matches session email,
 			// return the user to get extension data without authentication.
-			if ( ! empty( $_SERVER['HTTP_X_WOOPAY_VERIFIED_EMAIL_ADDRESS'] ) && ! empty( $session_data['woopay_email'] ) ) {
+			if ( ! empty( $_SERVER['HTTP_X_WOOPAY_VERIFIED_EMAIL_ADDRESS'] ) && ! empty( $customer['email'] ) ) {
```

Comment:
@alefesouza according to the comment above, which one is the "session email"? `$customer['email']` doesn't come from the WooPay session but the WC session right? Can we update the comment to clarify the intent of the code?

Reply:
We are settings the session customer email before redirection to WooPay [here](https://github.com/Automattic/woocommerce-payments/pull/6559/commits/300714bd9617fd8f407b56604c5ed52bc281b1e8#diff-57ee486c4a3ba4bae478eb31255c9a968b5b4eccd04b27f512e4003440d6f821R1510), with this we can do the the email verification and also fix 1951-gh-Automattic/woopay, I agree the comment should be updated.

Reply:
I like that we removed the extra cart token. 💯

I agree with Marcin's comment [here](https://github.com/Automattic/woopay/pull/1921#discussion_r1263653327). I think we could pass a boolean from WooPay if the email is verified or not, and handle the logic in the client.